### PR TITLE
JMeter samplers for transacted topics

### PR DESCRIPTION
**Pull request by riyafa**

### CODE REVIEW

**Comment by hemikak** [comment]
Out of curiosity, are these JMeter Samplers which will have  a GUI and all later on ?

**Comment by riyafa** [comment]
These are samplers for performance testing.

**Comment by hemikak** [comment]
👍

File: test_artifacts/jmeter-samplers/src/main/java/org/wso2/jmeter/java/samplers/TransactedTopicReceiver.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+
+/**
+ * This is Topic Receiver Jmeter Sampler class for local transactions.
+ */
+public class TransactedTopicReceiver extends AbstractJavaSamplerClient {
+    private static final Logger log = LoggingManager.getLoggerForClass();
+
+    private String userName;
```
Comment (pamod) [inline test_artifacts/jmeter-samplers/src/main/java/org/wso2/jmeter/java/samplers/TransactedTopicReceiver.java]: let's use javaDoc comments to describe these variables

File: test_artifacts/jmeter-samplers/src/main/java/org/wso2/jmeter/java/samplers/TransactedTopicReceiver.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        cfName = context.getParameter("CF_NAME");
+        properties = new Properties();
+        properties.put(Context.INITIAL_CONTEXT_FACTORY, context.getParameter("QPID_ICF"));
+        properties.put(context.getParameter("CF_NAME_PREFIX") +
+                               cfName, getTCPConnectionURL(userName, password, context));
+        properties.put(context.getParameter("TOPIC_NAME_PREFIX") + topicName, topicName);
+    }
+
+    @Override
+    public SampleResult runTest(JavaSamplerContext context) {
```
Comment (pamod) [inline test_artifacts/jmeter-samplers/src/main/java/org/wso2/jmeter/java/samplers/TransactedTopicReceiver.java]: javaDocs ?

File: test_artifacts/jmeter-samplers/src/main/java/org/wso2/jmeter/java/samplers/TransactedTopicReceiver.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+            result.setSuccessful(true);
+            result.setResponseMessage("Successfully consumed messages");
+            result.setResponseCodeOK();
+
+        } catch (JMSException | NamingException ex) {
+            result.sampleEnd();
+            result.setSuccessful(false);
+            // get stack trace as a String to return as document data
+            java.io.StringWriter stringWriter = new java.io.StringWriter();
+            ex.printStackTrace(new java.io.PrintWriter(stringWriter));
```
Comment (pamod) [inline test_artifacts/jmeter-samplers/src/main/java/org/wso2/jmeter/java/samplers/TransactedTopicReceiver.java]: Could  you explain the reason for logging the function as such ?

File: test_artifacts/jmeter-samplers/src/main/java/org/wso2/jmeter/java/samplers/TransactedTopicReceiver.java [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+            result.sampleEnd();
+            result.setSuccessful(true);
+            result.setResponseMessage("Successfully consumed messages");
+            result.setResponseCodeOK();
+
+        } catch (JMSException | NamingException ex) {
+            result.sampleEnd();
+            result.setSuccessful(false);
+            // get stack trace as a String to return as document data
+            java.io.StringWriter stringWriter = new java.io.StringWriter();
```
Comment (pamod) [inline test_artifacts/jmeter-samplers/src/main/java/org/wso2/jmeter/java/samplers/TransactedTopicReceiver.java]: Can we omit using fully qualified class paths ?

File: test_artifacts/jmeter-samplers/src/main/java/org/wso2/jmeter/java/samplers/TransactedTopicReceiver.java [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+            ex.printStackTrace(new java.io.PrintWriter(stringWriter));
+            result.setResponseData(stringWriter.toString(), null);
+            result.setResponseMessage("Unable to consume messages from topic." + "-" + "Exception: " + ex.toString());
+            result.setDataType(SampleResult.TEXT);
+            result.setResponseCode("FAILED");
+        }
+        return result;
+    }
+
+    private String getTCPConnectionURL(String username, String password, JavaSamplerContext context) {
```
Comment (pamod) [inline test_artifacts/jmeter-samplers/src/main/java/org/wso2/jmeter/java/samplers/TransactedTopicReceiver.java]: javaDocs ?

**Pull request closed without merging**